compiler/cpp/src/generate/t_cocoa_generator.cc

```diff
@@ -668,6 +669,40 @@ void t_cocoa_generator::generate_cocoa_struct_encode_with_coder_method(ofstream&
 }
 
 /**
+ * Generate the copyWithZone method for this struct so it's compatible with
+ * the NSCopying protocol
+ */
+void t_cocoa_generator::generate_cocoa_struct_copy_with_zone_method(ofstream& out,
+                                                                    t_struct* tstruct) {
+  indent(out) << "- (id) copyWithZone: (NSZone *) zone" << endl;
+  scope_up(out);
+  out << indent() << cocoa_prefix_ << tstruct->get_name()
+      << " *copy = [[[self class] allocWithZone: zone] init];" << endl;
+
+  const vector<t_field*>& members = tstruct->get_members();
+  vector<t_field*>::const_iterator m_iter;
+
+  for (m_iter = members.begin(); m_iter != members.end(); ++m_iter) {
+    t_type* t = get_true_type((*m_iter)->get_type());
+    out << indent() << "if (" << get_member_set_name((*m_iter)->get_name()) << ")" << endl;
+    scope_up(out);
+    if (type_can_be_null(t)) {
+      out << indent() << "copy->" << get_member_name((*m_iter)->get_name())
```

Comment:
maybe commit the result of `get_member_name((*m_iter)->get_name())` to a variable. It's repeated in a few places here.